File: frontend/types/work-done.ts
export interface WorkDone {
  id: number;
  clinic_id: number;
  work_name: string;
  created_at: string;
}

export interface WorkDoneCreate {
  work_name: string;
}

export interface WorkDoneUpdate {
  work_name: string;
}

export interface PatientWorkDone {
  id: number;
  clinic_id: number;
  patient_uid: string;
  patient_name: string;
  work_done_id?: number;
  description?: string;
  work_date?: string;
  created_at: string;
  work_name?: string;
}

export interface PatientWorkDoneCreate {
  patient_uid: string;
  patient_name: string;
  work_done_id?: number;
  description?: string;
  work_date?: string;
}

export interface PatientWorkDoneUpdate {
  patient_uid?: string;
  patient_name?: string;
  work_done_id?: number;
  description?: string;
  work_date?: string;
}

File: backend/app/schemas/work_done.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, datetime


class WorkDoneCreate(BaseModel):
    work_name: str


class WorkDoneUpdate(BaseModel):
    work_name: str


class WorkDoneResponse(BaseModel):
    id: int
    clinic_id: int
    work_name: str
    created_at: datetime

    class Config:
        from_attributes = True


class PatientWorkDoneCreate(BaseModel):
    patient_uid: str
    patient_name: str
    work_done_id: Optional[int] = None
    description: Optional[str] = None
    work_date: Optional[date] = None


class PatientWorkDoneUpdate(BaseModel):
    patient_uid: Optional[str] = None
    patient_name: Optional[str] = None
    work_done_id: Optional[int] = None
    description: Optional[str] = None
    work_date: Optional[date] = None


class PatientWorkDoneResponse(BaseModel):
    id: int
    clinic_id: int
    patient_uid: str
    patient_name: str
    work_done_id: Optional[int] = None
    work_name: Optional[str] = None
    description: Optional[str] = None
    work_date: Optional[date] = None
    created_at: datetime

    class Config:
        from_attributes = True
